Cabal/Distribution/Simple/SrcDist.hs

```diff
@@ -449,7 +449,7 @@ allSourcesBuildInfo verbosity bi pps modules = do
       in findFileWithExtension fileExts (hsSourceDirs bi) file
     | module_ <- modules ++ otherModules bi ]
 
-  return $ sources ++ catMaybes bootFiles ++ cSources bi ++ jsSources bi
+  return $ sources ++ catMaybes bootFiles ++ cSources bi ++ cxxSources bi ++ jsSources bi
```

Comment:
`cabal-testsuite/PackageTests/SDlist` has a test for `sdist`. Might be worth adding to?

Reply:
I figured out how to run the tests locally.